extras/userdoc/generator/modules/writers.py

```diff
@@ -0,0 +1,420 @@
+            f_file_name_hlp = open(('../cmds/sli/%s.hlp' % name), 'w')
+            f_file_name_hlp.write('\n'.join(hlplist))
+            f_file_name_hlp.close()
+        else:
+            f_file_name = open(('../cmds/cc/%s.html' % name), 'w')
+            f_file_name.write('\n'.join(htmllist))
+            f_file_name.close()
+            f_file_name_hlp = open(('../cmds/cc/%s.hlp' % name), 'w')
+            f_file_name_hlp.write('\n'.join(hlplist))
+            f_file_name_hlp.close()
+        # return name
+
+
+def write_helpindex(index_dic_list):
+    """
+    Write helpindex.html.
+
+    Collect the long list of dicts and transform it toa sorted html file.
+    """
+    head = '''
+    <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
+    <html>
+    <head>
+      <title>NEST Command Index</title>
+      <style type="text/css">
+        body {
+          padding: 0;
+          margin: 0;
+        }
+        h1 {
+          padding: 15px 0 0 15px;
+        }
+        p {
+          padding-left: 15px;
+        }
+        a {
+          color: #339;
+          text-decoration: none;
+        }
+        a:visited {
+          color: #339;
+          text-decoration: none;
+        }
+        a:hover {
+          text-decoration: underline;
+        }
+        h1 a {
+          color: #000;
+          text-decoration: none;
+        }
+        table.headerfooter {
+          margin: 20px 0 20px 0;
+          background-color: #eee;
+          width: 100%;
+          height: 30px;
+          border-top: 2px solid #ccc;
+          border-bottom: 2px solid #ccc;
+          text-align: center;
+        }
+        table.commands {
+          margin: 15px 0 15px 0;
+          background-color: #eee;
+          width: 90%;
+          border: 2px solid #ccc;
+          border-spacing: 0px;
+          border-collapse: collapse;
+        }
+        table.commands td {
+          border-bottom: 1px solid #ccc;
+          border-right: 1px dotted #ccc;
+          padding: 5px 0 5px 10px;
+          text-align: left;
+        }
+        table.letteridx {
+          margin: 0;
+          background-color: #eee;
+          width: 90%;
+          border: 2px solid #ccc;
+          border-spacing: 0px;
+          border-collapse: collapse;
+        }
+        table.letteridx td {
+          border-right: 1px solid #ccc;
+          padding: 5px;
+          text-align: center;
+        }
+        table.letteridx a {
+          display: block;
+          height: 100%;
+          width: 100%;
+        }
+        td.left{
+            width:30%;
+        }
+      </style>
+    </head>
+    <body bgcolor="white" fgcolor="black">
+    <h1><a name="top">NEST Command Index</a></h1>
+    <table class="headerfooter">
+      <tr>
+        <td width="50%" align=center>
+        <a href="../index.html">NEST HelpDesk</a></td>
+        <td width="50%" align=center>
+        <a href="../quickref.html">NEST Quick Reference</a></td>
+      </tr>
+    </table>
+    '''
```

Comment:
I thing we should use a template engine here to increase code readability and maintainability. [string.Template](https://docs.python.org/2/library/string.html#template-strings) is in the Python standard library and should suffice for this.

I also vote for putting the different blocks in separate files called `nest.css`, `header.html`, `footer.html` and loading them here. This way they can be easily edited separately with an appropriate editor, syntax highlighting and all.